packages/liveblocks-client/src/__tests__/room.test.ts

```diff
@@ -848,6 +868,30 @@ describe("room", () => {
 
       const itemsSubscriber = jest.fn();
       const refItemsSubscriber = jest.fn();
+
+      //
+      // XXX This unit test is failing.
+      //
+      // Here's what I understand so far:
+      // - We're trying to keep track of the last "others" callback message
+      //   called (in the ref machine), in the `refOthers` variable.
+      // - In this now-failing unit test, that callback doesn't get called
+      //   correctly anymore: `refPresenceSubscriber` is not getting invoked.
+      // - This is somewhat expected, because in order to get `others`
+      //   callbacks, the machine will have to have _all_ initial presence data
+      //   first.
+      // - So, before the updatePresence() calls can do anything, the ref
+      //   machine will first need to have been "properly connected" to the
+      //   main machine.
+      // - This means the ref machine should first receive either a ROOM_STATE
+      //   or a USER_JOINED message from the server.
+      // - This doesn't seem to be happening. Very possibly this is a detail
+      //   that isn't implemented in the mock implementation of the ref machine.
+      // - I tried calling `refMachine.connect()`, but this didn't make
+      //   a difference.
+      //
+      // - Help?
+      //
```

Comment:
This is a tricky one. Perhaps @ofoucherot you know enough about this test setup to help me solve this issue? I've documented my findings here in this comment. Would love to get some help from you on this one! 🙏

Reply:
The util `prepareStorageTest` mocks 2 users connected to a room (one user and one user ref). It doesn’t reproduce the server logic completely, basically, it just forwards storage updates to the other user. We don’t want to replicate the full server logic in the unit tests, if it’s needed we can use the integration tests. If we want to unit test custom flow, we need to send server messages manually.